src/error.rs

```diff
@@ -167,6 +168,15 @@ impl<'src> Error<'src> {
       message: message.into(),
     }
   }
+
+  pub(crate) fn suppress_message(&self) -> bool {
+    match self {
+      Error::Code {
+        suppress_message, ..
+      } => *suppress_message,
+      _ => false,
+    }
+  }
```

Comment:
I'm not 100% sure this is the right syntax, but it should be possible to simplify this like so:

```suggestion
    matches!(self, Error::Code { suppress_message: true, ..})
```